packages/compass-indexes/src/modules/regular-indexes.ts

```diff
@@ -182,6 +182,17 @@ export const INITIAL_STATE: State = {
   error: undefined,
 };
 
+function processError(error: string): string {
+  const internalCodePattern = /[A-Z_]+:/; // Matches all caps with underscores followed by a colon anywhere in the string
+  const httpCodePattern = /\b(4\d{2}|5\d{2})\b/; // Matches HTTP codes 400-599 anywhere in the string
+
+  if (internalCodePattern.test(error) || httpCodePattern.test(error)) {
+    return 'Sorry, an unexpected error has occurred. Try refreshing the page or reporting the issue if it persists.';
```

Comment:
Nitpick but let's keep it as consistent with ye olde DE as possible

"We're sorry, an unexpected error has occurred. Please try again."